autode/species/species.py

```diff
@@ -533,22 +556,46 @@ def _generate_conformers(self, *args, **kwargs):
         raise NotImplementedError('Could not generate conformers. '
                                   'generate_conformers() not implemented')
 
-    def _run_hess_calculation(self, method):
-        """Run a Hessian calculation on this species"""
-
-        if self.n_atoms < 2:
-            logger.warning(f'Not running a Hessian calculation on only '
-                           f'{self.n_atoms} atom(s). Cannot have frequencies')
-            return None
+    def _default_hessian_calculation(self, method=None, keywords=None):
+        """Construct a default Hessian calculation"""
 
         method = method if method is not None else get_hmethod()
+        keywords = keywords if keywords is not None else method.keywords.hess
 
         calc = Calculation(name=f'{self.name}_hess',
                            molecule=self,
                            method=method,
-                           keywords=method.keywords.hess,
+                           keywords=keywords,
                            n_cores=Config.n_cores)
+
+        return calc
+
+    def _run_hess_calculation(self, **kwargs):
+        """Run a Hessian calculation on this species
+
+        ----------------------------------------------------------------------
+        Keyword Arguments:
+            calc: Calculation, if undefined then use a default calculation
+
+            method: Method to use for the calculation, if it's undefined.
+                    Defaults to methods.get_hmethod()
+
+            keywords: Keywords to use in a calculation, if it's undefined.
+                      Defaults to method.keywords.hess
+        """
+
+        if self.n_atoms < 2:
+            logger.warning(f'Not running a Hessian calculation on only '
+                           f'{self.n_atoms} atom(s). Cannot have frequencies')
+            return None
+
+        calc = kwargs.pop('calc', None)
```

Comment:
idk if this is any better, but `calc = kwargs.pop('calc', self._default_hessian_calculation(**kwargs))` ?

Reply:
I thought about this but the signature of `_default_hessian_calculation` I don't think allows for it, as the calc needs to be pop-ed out of the kwargs that are then passed to `_default_hessian_calculation `(?)